clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultBackgroundThread.java

```diff
@@ -51,153 +40,47 @@
  * It holds a reference to the {@link SubscriptionState}, which is
  * initialized by the polling thread.
  */
-public class DefaultBackgroundThread extends KafkaThread {
+public class DefaultBackgroundThread<K, V> implements Runnable, Closeable {
```

Comment:
Curious about the motive for having the DefaultBackgroundThread as a Runnable instead of a KafkaThread. Having it as runnable that is later on supplied to a new KafkaThread is definitely a valid alternative, but I feel like it introduces complexity in the event handler, that ends up having to keep 2 things for the same purpose: an instance of the defaultBackgroundThread and a KafkaThread to run it in. It then has to deal with both in places like the event handler close. 

I might be missing the wins of this new approach but wondering if having the defaultBackgroundThread as a thread (extending KafkaThread) simplifies its management

Reply:
Part of it was to mimic what is done on the producer side. `KafkaProducer` instantiates a `Sender` (a `Runnable`) and then creates a `KafkaThread` around that.

Secondly, I was getting errors in the checker when the `DefaultBackgroundThread` created a thread for itself and then invoked `start()` on itself.

But I agree that we should discuss more and probably have a PR more or less dedicated to that change so that it's called out more explicitly.

Reply:
Reverted to having `DefaultBackgroundThread` extend from `KafkaThread` directly. Seems fine now 🤷‍♂️